### ISSUE TITLE
Change Branch Lock Policy to look for Tag on Bug Work Items

### ISSUE THREAD

**Issue by MrHinsh** [nkdAgility/TfvcBranchPolicy#5]
During the evaluation of the Branch Lock Policy the bypasspattern is looked for only in the Comment provided. This should be changed to look first to see if this bypasspattern has been added to any of the work items associated with the checkin, and fall back to the Comment if it is not.

**Comment by hmahajanHM** [comment]
Not clear what to do with these tags identified on Bug WI?

**Comment by MrHinsh** [comment]
Good point...i have updated the requirement. Also please merge the latest changes before starting...

**Comment by MrHinsh** [comment]
Do you think that there is more value in iterating through the history than just the Tags (WorkItem.Tags)?

I am concerned about the performance in production...

**Comment by hmahajanHM** [comment]
The dll Microsoft.TeamFoundation.WorkItemTracking.Client version added in TfvcBranchPolicy.CheckinPolicy project does not support Tags in WI. 

Or we can restrict the History comments to recent 5?

**Comment by MrHinsh** [comment]
I have access to Tags with the 2013 version of the API that we are using:
![image](https://cloud.githubusercontent.com/assets/5205575/12120753/708b4ace-b3d2-11e5-9cf8-b41f286ba8ce.png)
**Can you make sure that you have Visual Studio 2013 Update 5 installed**

**Comment by hmahajanHM** [comment]
Ok let me check at my end again

**Comment by hmahajanHM** [comment]
This week I will be able to spend approx. 5-6 hours on this tool.

**Comment by MrHinsh** [comment]
That sounds awesome... :) did you see that I commented on the code for https://github.com/hmahajanHM/TfvcBranchPolicy/commit/0beb59d20d32d36e8a2dc394a6b50cbca9c5867e

**Comment by hmahajanHM** [comment]
yes, was not sure how tags treet ; so thought better to rely on TFS for and break tags separate and then check each one.

**Comment by MrHinsh** [comment]
How did you get on with this?
